transactional_licence_form/views.py

```diff
@@ -58,22 +49,30 @@ def post(self, *args, **kwargs):
 
         form = self.get_form(data=self.request.POST, files=self.request.FILES)
 
-        if form.is_valid():
-            self.storage.set_step_data(self.steps.current, self.process_step(form))
-            self.storage.set_step_files(
-                self.steps.current, self.process_step_files(form)
-            )
+        wizard_goto_step = self.request.POST.get("wizard_goto_step", None)
 
+        # If we're in review or pressing 'Next', we need to ensure the user fixes bad data
+        if not form.is_valid() and (self.in_review() or wizard_goto_step is None):
+            return self.render(form)
+
+        self.storage.set_step_data(self.steps.current, self.process_step(form))
+        self.storage.set_step_files(self.steps.current, self.process_step_files(form))
+
+        # Going to a step out of order takes priority
+        if wizard_goto_step and wizard_goto_step in self.get_form_list():
+            return self.render_goto_step(wizard_goto_step)
+
+        # Since the user isn't jumping, they must be moving forwards
+        if form.is_valid():
```

Comment:
We could continue the style of single `if` statements and remove the `else` later on by making this:

```pythong
if not form.is_valid():
    return self.render(form)
```

then we just follow up with:

```
if self.steps.current == self.steps.last:
    return self.render_done(form, **kwargs)
```

Finally:

```
return self.render_next_step(form)
```